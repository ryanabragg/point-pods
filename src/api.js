import localforage from 'localforage';
import React from 'react';

const api = {
  store: localforage.createInstance({
    name: 'Point Pods',
    storeName: 'point_pods',
    description: 'some description'
  }),
};

export default api;
export const APIContext = React.createContext(api);
export function withAPI(Component) {
  return function APIComponent(props) {
    return (
      <APIContext.Consumer>
        {context => <Component {...props} api={context} />}
      </APIContext.Consumer>
    );
  };
}
