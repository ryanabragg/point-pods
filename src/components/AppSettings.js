import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { withAPI } from '../api';

import Divider from '@material-ui/core/Divider';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import TextField from '@material-ui/core/TextField';

const styles = theme => ({
  root: {
    maxWidth: theme.breakpoints.values['sm'],
  },
  content: {
    padding: theme.spacing.unit,
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-evenly',
    alignItems: 'flex-start',
    alignContent: 'flex-start',
  },
  field: {
    minWidth: 250,
  },
});

class AppSettings extends Component {
  state = {
    pairingMethod: '',
    pairingMethodInitial: '',
    podSizeMinimum: '',
    podSizeMaximum: '',
    pairingMethods: [],
  };

  componentDidMount() {
    let loadState = {
      pairingMethods: Object.keys(this.props.api.pairingMethods)
        .map(key => ({
          label: key.slice(0,1).toUpperCase() + key.slice(1).toLowerCase(),
          value: this.props.api.pairingMethods[key],
        })),
    };
    this.props.api.Settings.get()
      .then((settings) => Object.assign(loadState, settings))
      .then(() => this.setState(loadState))
      .catch(error => this.props.notification(error.message, 'error'));
  }

  handleInputChange = name => event => {
    const value = event.target.value;
    this.props.api.Settings.set({ [name]: value })
      .then(settings => this.setState(settings))
      .catch(error => this.props.notification(error.message, 'error'));
  };

  render() {
    const { classes } = this.props;
    const {
      pairingMethod,
      pairingMethodInitial,
      podSizeMinimum,
      podSizeMaximum,
      pairingMethods,
    } = this.state;
    return (
      <Card className={classes.root}>
        <CardHeader
          title='Pod Setup'
        />
        <Divider />
        <CardContent className={classes.content}>
          <TextField className={classes.field}
            margin="dense"
            id="settings-pairingMethod"
            label="Pairing Method"
            type="text"
            value={pairingMethod}
            onChange={this.handleInputChange('pairingMethod')}
            select
          >
            {pairingMethods.map(method => (
              <option key={method.value} value={method.value}>
                {method.label}
              </option>
            ))}
          </TextField>
          <TextField className={classes.field}
            margin="dense"
            id="settings-pairingMethodInitial"
            label="Initial Pairing Method"
            type="text"
            value={pairingMethodInitial}
            onChange={this.handleInputChange('pairingMethodInitial')}
            select
          >
            {pairingMethods.map(method => (
              <option key={method.value} value={method.value}>
                {method.label}
              </option>
            ))}
          </TextField>
          <TextField className={classes.field}
            margin="dense"
            id="settings-podSizeMinimum"
            label="Minimum Pod Size"
            type="number"
            value={podSizeMinimum}
            onChange={this.handleInputChange('podSizeMinimum')}
          />
          <TextField className={classes.field}
            margin="dense"
            id="settings-podSizeMaximum"
            label="Maximum Pod Size"
            type="number"
            value={podSizeMaximum}
            onChange={this.handleInputChange('podSizeMaximum')}
          />
        </CardContent>
      </Card>
    );
  }
}

AppSettings.defaultProps = {
  notification: (message, variant, duration, onClose) => null,
};

AppSettings.propTypes = {
  api: PropTypes.object.isRequired, // added by withAPI
  classes: PropTypes.object.isRequired, // added by withStyles
  notification: PropTypes.func,
};

export default withStyles(styles, { withTheme: true })(withAPI(AppSettings));
