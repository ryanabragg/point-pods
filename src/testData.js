import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
