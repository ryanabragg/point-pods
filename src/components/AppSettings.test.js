import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CardHeader from '@material-ui/core/CardHeader';

import AppSettings from './AppSettings';

import api from '../api';

import { testSettings, testTournaments, mockStorage, mockAPI, demockAPI } from '../testData';

beforeEach(() => mockAPI(api, mockStorage));
afterEach(() => demockAPI(api));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppSettings />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', (done) => {
  const component = renderer.create(<AppSettings />);
  expect(component.toJSON()).toMatchSnapshot();
    component.update(<AppSettings />);
  setTimeout(() => {
    try {
      expect(component.toJSON()).toMatchSnapshot();
      done();
    } catch (error) {
      done.fail(error);
    }
  }, 100);
});

test('is a HOC-ed component', () => {
  const component = shallow(<AppSettings />);
  expect(component.type().prototype.constructor.name).toBe('APIComponent');
  expect(component.dive().type().prototype.constructor.name).toBe('AppSettings');
});

describe('default values', () => {
  test('loads the pod sorting options', (done) => {
    let methodsArray = Object.keys(api.pairingMethods)
      .map(key => ({
        label: key.slice(0,1).toUpperCase() + key.slice(1).toLowerCase(),
        value: api.pairingMethods[key]
      }));
    const component = shallow(<AppSettings />);
    const unHOC = component.dive().dive();
    setTimeout(() => {
      try {
        expect(unHOC.state('pairingMethods')).toEqual(methodsArray);
        done();
      } catch (error) {
        done.fail(error);
      }
    }, 100);
  });

  test('loads the api tournament settings', (done) => {
    const component = shallow(<AppSettings />);
    const unHOC = component.dive().dive();
    setTimeout(() => {
      try {
        expect(unHOC.state('pairingMethod')).toBe(testSettings.pairingMethod);
        expect(unHOC.state('pairingMethodInitial')).toBe(testSettings.pairingMethodInitial);
        expect(unHOC.state('podSizeMinimum')).toBe(testSettings.podSizeMinimum);
        expect(unHOC.state('podSizeMaximum')).toBe(testSettings.podSizeMaximum);
        done();
      } catch (error) {
        done.fail(error);
      }
    }, 100);
  });

  test('saves settings as they are changed', (done) => {
    const spy = jest.fn(r => r);
    const component = shallow(<AppSettings />);
    const unHOC = component.dive().dive();
    setTimeout(() => {
      try {
        expect(testSettings.pairingMethod).toBe(api.pairingMethods.HISTORIC);
        unHOC.find('#settings-pairingMethod').simulate('change', {target: {value: api.pairingMethods.POINTS}});
        expect(testSettings.pairingMethodInitial).toBe(api.pairingMethods.HISTORIC);
        unHOC.find('#settings-pairingMethodInitial').simulate('change', {target: {value: api.pairingMethods.RANDOM}});
        expect(testSettings.podSizeMinimum).toBe(4);
        unHOC.find('#settings-podSizeMinimum').simulate('change', {target: {value: 3}});
        expect(testSettings.podSizeMaximum).toBe(7);
        unHOC.find('#settings-podSizeMaximum').simulate('change', {target: {value: 4}});
        setTimeout(() => {
          try {
            expect(testSettings.pairingMethod).toBe(api.pairingMethods.POINTS);
            expect(testSettings.pairingMethodInitial).toBe(api.pairingMethods.RANDOM);
            expect(testSettings.podSizeMinimum).toBe(3);
            expect(testSettings.podSizeMaximum).toBe(4);
            done();
          } catch (error) {
            done.fail(error);
          }
        }, 100);
      } catch (error) {
        done.fail(error);
      }
    }, 100);
  });
});
