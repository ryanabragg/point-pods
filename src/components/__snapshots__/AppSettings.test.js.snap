// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div
  className="MuiPaper-root-115 MuiPaper-elevation1-118 MuiPaper-rounded-116 MuiCard-root-114 APIComponent-root-111"
>
  <div
    className="MuiCardHeader-root-142"
  >
    <div
      className="MuiCardHeader-content-145"
    >
      <span
        className="MuiTypography-root-148 MuiTypography-headline-153 MuiCardHeader-title-146"
      >
        Pod Setup
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root-173"
  />
  <div
    className="MuiCardContent-root-177 APIComponent-content-112"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-marginDense-184"
        data-shrink={false}
        htmlFor="settings-pairingMethod"
      >
        Pairing Method
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <div
          className="MuiSelect-root-192"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-193 MuiSelect-selectMenu-194 MuiInput-input-205 MuiInput-inputMarginDense-206"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="settings-pairingMethod"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-210 MuiSelect-icon-196"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-marginDense-184"
        data-shrink={false}
        htmlFor="settings-pairingMethodInitial"
      >
        Initial Pairing Method
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <div
          className="MuiSelect-root-192"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-193 MuiSelect-selectMenu-194 MuiInput-input-205 MuiInput-inputMarginDense-206"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="settings-pairingMethodInitial"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-210 MuiSelect-icon-196"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-marginDense-184"
        data-shrink={false}
        htmlFor="settings-podSizeMinimum"
      >
        Minimum Pod Size
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <input
          aria-invalid={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-205 MuiInput-inputType-208 MuiInput-inputMarginDense-206"
          defaultValue={undefined}
          disabled={false}
          id="settings-podSizeMinimum"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={false}
          rows={undefined}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-marginDense-184"
        data-shrink={false}
        htmlFor="settings-podSizeMaximum"
      >
        Maximum Pod Size
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <input
          aria-invalid={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-205 MuiInput-inputType-208 MuiInput-inputMarginDense-206"
          defaultValue={undefined}
          disabled={false}
          id="settings-podSizeMaximum"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={false}
          rows={undefined}
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div
  className="MuiPaper-root-115 MuiPaper-elevation1-118 MuiPaper-rounded-116 MuiCard-root-114 APIComponent-root-111"
>
  <div
    className="MuiCardHeader-root-142"
  >
    <div
      className="MuiCardHeader-content-145"
    >
      <span
        className="MuiTypography-root-148 MuiTypography-headline-153 MuiCardHeader-title-146"
      >
        Pod Setup
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root-173"
  />
  <div
    className="MuiCardContent-root-177 APIComponent-content-112"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-shrink-185 MuiInputLabel-marginDense-184"
        data-shrink={true}
        htmlFor="settings-pairingMethod"
      >
        Pairing Method
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <div
          className="MuiSelect-root-192"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-193 MuiSelect-selectMenu-194 MuiInput-input-205 MuiInput-inputMarginDense-206"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Historic
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="settings-pairingMethod"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value={1}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-210 MuiSelect-icon-196"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-shrink-185 MuiInputLabel-marginDense-184"
        data-shrink={true}
        htmlFor="settings-pairingMethodInitial"
      >
        Initial Pairing Method
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <div
          className="MuiSelect-root-192"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-193 MuiSelect-selectMenu-194 MuiInput-input-205 MuiInput-inputMarginDense-206"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Historic
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="settings-pairingMethodInitial"
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value={1}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-210 MuiSelect-icon-196"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-shrink-185 MuiInputLabel-marginDense-184"
        data-shrink={true}
        htmlFor="settings-podSizeMinimum"
      >
        Minimum Pod Size
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <input
          aria-invalid={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-205 MuiInput-inputType-208 MuiInput-inputMarginDense-206"
          defaultValue={undefined}
          disabled={false}
          id="settings-podSizeMinimum"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={false}
          rows={undefined}
          type="number"
          value={4}
        />
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-178 MuiFormControl-marginDense-180 APIComponent-field-113"
    >
      <label
        className="MuiFormLabel-root-187 MuiInputLabel-root-182 MuiInputLabel-formControl-183 MuiInputLabel-animated-186 MuiInputLabel-shrink-185 MuiInputLabel-marginDense-184"
        data-shrink={true}
        htmlFor="settings-podSizeMaximum"
      >
        Maximum Pod Size
      </label>
      <div
        className="MuiInput-root-197 MuiInput-formControl-198 MuiInput-underline-201"
      >
        <input
          aria-invalid={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-205 MuiInput-inputType-208 MuiInput-inputMarginDense-206"
          defaultValue={undefined}
          disabled={false}
          id="settings-podSizeMaximum"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={false}
          rows={undefined}
          type="number"
          value={7}
        />
      </div>
    </div>
  </div>
</div>
`;
